cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -105,68 +112,77 @@ public CruiseControlApiImpl(Vertx vertx, int idleTimeout) {
 
         return HttpClientUtils.withHttpClient(vertx, options, (httpClient, result) -> {
-            HttpClientRequest request = httpClient.post(port, host, path, response -> {
-                response.exceptionHandler(result::tryFail);
-                if (response.statusCode() == 200 || response.statusCode() == 201) {
-                    response.bodyHandler(buffer -> {
-                        String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
-                        JsonObject json = buffer.toJsonObject();
-                        CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
-                        result.complete(ccResponse);
-                    });
-                } else if (response.statusCode() == 202) {
-                    response.bodyHandler(buffer -> {
-                        String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
-                        JsonObject json = buffer.toJsonObject();
-                        CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
-                        if (json.containsKey(CC_REST_API_PROGRESS_KEY)) {
-                            // If the response contains a "progress" key then the rebalance proposal has not yet completed processing
-                            ccResponse.setProposalStillCalaculating(true);
-                        } else {
-                            result.fail(new CruiseControlRestException(
-                                    "Error for request: " + host + ":" + port + path +
-                                            ". 202 Status code did not contain progress key. Server returned: " +
-                                            ccResponse.getJson().toString()));
-                        }
-                        result.complete(ccResponse);
-                    });
-                } else if (response.statusCode() == 500) {
-                    response.bodyHandler(buffer -> {
-                        String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
-                        JsonObject json = buffer.toJsonObject();
-                        if (json.containsKey(CC_REST_API_ERROR_KEY)) {
-                            // If there was a client side error, check whether it was due to not enough data being available
-                            if (json.getString(CC_REST_API_ERROR_KEY).contains("NotEnoughValidWindowsException")) {
-                                CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
-                                ccResponse.setNotEnoughDataForProposal(true);
-                                result.complete(ccResponse);
+            httpClient.request(HttpMethod.POST, port, host, path, request -> {
+                if (request.succeeded()) {
+                    if (idleTimeout != HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS) {
```

Comment:
I might be wrong but here you seem to be setting the idle timeout for a request that has already suceeded and returned?

Reply:
The httpClient.request docs say that the method:
Create an HTTP request to send to the server at the default host and port. The handler is called when the request is ready to be sent.

So this might be right, I am just confused as to what `request.succeeded()` means here?

Reply:
The last argument of `request` method is `Handler<AsyncResult<HttpClientRequest>>` so it's result is a `HttpClientRequest` object.

Reply:
Ah right it is actually the `AsyncResult` that contains the request you are calling `.suceeded()` on. Ok.